test/unit/php/includes/core/classes/endpoints/class-test-endpoint-template.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Class handles unit tests for GatherPress\Core\Endpoints\Endpoint_Template.
+ *
+ * @package GatherPress\Core\Endpoints
+ * @since 1.0.0
+ */
+
+namespace GatherPress\Tests\Core;
+
+use GatherPress\Core\Endpoints\Endpoint_Template;
+use PMC\Unit_Test\Base;
+use PMC\Unit_Test\Utility;
+
+/**
+ * Class Test_Endpoint_Template.
+ *
+ * @coversDefaultClass \GatherPress\Core\Endpoints\Endpoint_Template
+ * @group              endpoints
+ */
+class Test_Endpoint_Template extends Base {
+
+	/**
+	 * Coverage for activate.
+	 *
+	 * @covers ::__construct
+	 * @covers ::activate
+	 *
+	 * @return void
+
+	public function test_activate(): void {
+		$cb = function(){};
+		$instance = new Endpoint_Template( 'unit-test', $cb );
+		$hooks    = array(
+			array(
+				'type'     => 'filter',
+				'name'     => 'template_include',
+				'priority' => 10,
+				'callback' => array( $instance, 'template_include' ),
+			),
+		);
+
+		$this->assert_hooks( $hooks, $instance );
+	} */
```

Comment:
Following how other hooks are tested, I tried to test if an endpoint gets hooked onto `template_include`, but I wasn't able to get this test running.